pkg/utils/net-attach-def.go

```diff
@@ -152,14 +152,41 @@ func CreateNetworkStatuses(r cnitypes.Result, networkName string, defaultNetwork
 	// Same for DNS
 	v1dns := convertDNS(result.DNS)
 
+	// Check for a gateway-associated interface, we'll use this later if we did to mark as the default.
+	gatewayInterfaceIndex := -1
+	if defaultNetwork {
+		for _, ipConfig := range result.IPs {
+			if ipConfig.Gateway != nil && ipConfig.Interface != nil {
+				// Keep the index of the first interface that has a gateway
+				gatewayInterfaceIndex = *ipConfig.Interface
+				break
+			}
+		}
+	}
```

Comment:
nit: maybe, for readability, you should export this into an helper ? 

Something like:
```suggestion
        if defaultNetwork {
                gatewayIfaceIndex := gatewayInterfaceIndex(result.IPs)
        }
```

And the helper would be something like:
```golang
func gatewayInterfaceIndex(ips IPs) int {
        for _, ipConfig := range ips {
			if ipConfig.Gateway != nil && ipConfig.Interface != nil {
				return *ipConfig.Interface
			}
		}
	return -1
}
```

I do wonder about CNIs that don't bother setting the `Interface` attribute for IPs in the CNI result ... 

We would discard them; is that OK ? I think so - we can always open a bug on them to honor the CNI spec.

Reply:
re: 

Totally hear that concern, I think that we were kind of in that situation previously too with the existing:

```
for i, iface := range result.Interfaces {
```

Buuuut... You're right. There's a sticky situation here with those in that if you don't set any interfaces, you don't have anything to index in the IPs either, so we can't determine if there's a gateway, and we can't determine if it's the first sandbox interface either. Sooo... I will try to do my best to explain what needs to be done in the release note too, and potentially update the language in the spec update, too.

Helper function++, can't hurt to break it out, makes this block easier to read for the main functionality.